core/src/main/scala/latis/input/TextAdapter.scala

```diff
@@ -39,23 +33,13 @@ class TextAdapter(model: DataType, config: TextAdapter.Config = new TextAdapter.
   
       
   /**
-   * Given a line of text from the data source, return a Boolean
-   * expressing whether we are in the state of seeking the data
-   * marker that indicates that the data starts on the next line.
-   * This is used by the record Stream to drop lines that do not 
-   * need to be parsed.
+   * Given a data marker that indicates that the data starts on
+   * the next line, drop elements of the stream until one matches
+   * the marker. Also drop the line with the marker.
    */
-  private val seekingDataMarker: String => Boolean = (line: String) => {
-    var foundDataMarker = false
-    config.dataMarker match {
-      case Some(marker) => {
-        //a little gymnastics to also skip the line with the marker (TODO: the line with the marker isn't getting skipped...)
-        //val seeking = ! foundDataMarker
-        if (line.matches(marker)) foundDataMarker = true
-        !foundDataMarker //seeking
-      }
-      case None => false //no marker defined so we are not seeking
-    }
+  private def seekToDataMarker(marker: Option[String]): Pipe[IO, String, String] = marker match {
+    case Some(m) => _.dropWhile(! _.matches(m)).drop(1)
+    case None => s: Stream[IO, String] => s
```

Comment:
There is an `Identity` that we could try here.

Reply:
Yeah, this is apparently a thing:

```suggestion
    case None => identity
```

Reply:
@dlindhol I'm leaving it to you to try that `Identity`. That's beyond me. lol

Reply:
Oh! I didn't realize it was as simple as typing `identity`. That's super cool, and it totally works. I tested it thoroughly.

Reply:
Code updated.